cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/distributed/SparkContainerLauncher.java

```diff
@@ -207,4 +201,29 @@ public void run() {
   private static boolean isPySpark() {
     return System.getenv("PYTHONPATH") != null;
   }
+
+  /**
+   * Removes the first jar file containing the given class based on the ordering providing by
+   * the given {@link ClassLoader} if there are multiple jars containing the given class.
+   *
+   * @param classLoader the {@link ClassLoader} for finding the jar containing the given class
+   * @param className the class to look for
+   * @param urls a {@link Set} of {@link URL} for the removal jar file
+   * @throws IOException if failed to lookup the class location
+   */
+  private static void removeFirstJar(ClassLoader classLoader, String className, Set<URL> urls) throws IOException {
+    // Remove the first scala from the set of classpath. This ensure the one from Spark is used for spark
```

Comment:
this comment is too specific for the code that it is commenting, right?

Reply:
fixed